processing/src/main/java/org/apache/druid/query/QueryContexts.java

```diff
@@ -503,7 +487,24 @@ public static String getAsString(
     } else if (value instanceof String) {
       return (String) value;
     } else {
-      throw new IAE("Expected parameter [%s] to be String", parameter);
+      throw new IAE("Expected parameter [%s] to be a String, but got [%s]", parameter, value.getClass().getName());
+    }
+  }
+
+  @Nullable
+  public static Boolean getAsBoolean(
+      final String parameter,
+      final Object value
```

Comment:
This set of methods has gotten a bit muddled.

* This version takes the name of a context key (not "parameter") just so it can throw an error. But, there are places where we (or at least I) want to use this method where the map is not a context.
* There are versions that take a map and a default, and versions, like this, that take a value (not parameter) and convert it.

Maybe have the following set:

```java
// Check if a key exists for code that wants to do something special for this case
public boolean hasValue(Map<..> context, String key) ...

// Generic method to convert objects to boolean
public boolean parseBoolean(Object value)
{
    // if can't parse, throw an exception
}

// Specific method for boolean context values
public boolean getBoolean(Map<...> context, String key, boolean defaultValue)
{
  value = context.get(key);
  if (value == null) {
    return defaultValue;
  }
  try {
      return parseBoolean(value);
   } catch (Exception e) {
     // Throw the "Expected context value [%s] to be a ... exception
  }
}
```

```

Reply:
A `parseBoolean` is still kept at line 474 in this class to allow to parse a value from a map object.